### PR TITLE
allow publishing testkit

### LINKED ISSUE #11529: testkit isn't published
nightlies for partest are unusable, as they depend on scala-testkit, which isn't published.

### PR DESCRIPTION
**Pull request by martijnhoekstra**
Don't skip publishing testkit, it's a dependency for partest

Fixes https://github.com/scala/bug/issues/11529

### CODE REVIEW

**Comment by dwijnand** [comment]
If this pops up in https://scala-ci.typesafe.com/artifactory/scala-pr-validation-snapshots/org/scala-lang/ then we're golden.

**Comment by martijnhoekstra** [comment]
I have no idea what I'm doing

**Comment by martijnhoekstra** [comment]
https://scala-ci.typesafe.com/artifactory/scala-pr-validation-snapshots/org/scala-lang/scala-testkit/

there it is

**Comment by hrhino** [comment]
> I have no idea what I'm doing

@martijnhoekstra does anyone? Well, Dale, I suppose.

File: scripts/jobs/validate/publish-core [hunk 1]
```diff
@@ -28,7 +28,7 @@ case $prDryRun in
     if $libraryAvailable && $reflectAvailable && $compilerAvailable; then
       echo "Scala core already built!"
     else
-      $SBT -warn "setupPublishCore $prRepoUrl" publish
+      $SBT "setupPublishCore $prRepoUrl" publish
```
Comment (dwijnand) [inline scripts/jobs/validate/publish-core]: Let's revert this.

And then squash it up.

**Comment by hrhino** [comment]
Should this make the RC2 cut?

**Comment by martijnhoekstra** [comment]
I want it for snapshot testing (so I dont really care, as long as RC2 is soon, which it probably is). Im not sure anyone else cares.

While I think this is riskless, that sounds like famous last words. So let's not.

**Review by dwijnand — APPROVED** [review]

**Comment by dwijnand** [comment]
`#good-to-go-after-2.13.0-RC2`

**Comment by dwijnand** [comment]
Rebased the merge conflict (we lost that compiler options exporter project).  This is good to go when green.

**Pull request merged**